bin/hostname/tests/hostname_test.sh

```diff
@@ -0,0 +1,57 @@
+#-
+# Copyright (c) 2023 Klara, Inc.
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+#
+# These tests need to run in a multibyte locale with non-localized
+# error messages.
+#
+export LC_CTYPE=C.UTF-8
+export LC_MESSAGES=C
+
+root_dir=$(atf_get_srcdir)
+
+test_jail_conf='test-hostname-jail {
+    host.hostname = "test-hostname";
+    path = "/";
+    persist;
+}'
+
+init()
+{
+    echo "$test_jail_conf" > "$root_dir/jail.conf"
+    jail -f "$root_dir/jail.conf" -c test-hostname-jail
```

Comment:
I suggest extracting `test-hostname-jail` to an variable, as it's used multiple times in this file.